packages/integration-karma/test/rendering/slotted-text-diffing/index.spec.js

```diff
@@ -0,0 +1,24 @@
+import { createElement } from 'lwc';
+import Container from 'x/container';
+
+describe('Dynamic diffing algo for slotted text', () => {
+    it('should not confuse text vnodes when moving elements', function () {
+        const elm = createElement('x-container', { is: Container });
+        document.body.appendChild(elm);
+
+        elm.toggleVisibility();
+
+        return Promise.resolve()
+            .then(() => {
+                const text = elm.shadowRoot.textContent;
+                expect(text).toContain('false');
```

Comment:
We should match against the entire text content here to make sure that all the text is in the right place, not only the partial content.